heron/spi/src/java/com/twitter/heron/spi/statemgr/SchedulerStateManagerAdaptor.java

```diff
@@ -209,6 +209,31 @@ public Boolean deleteSchedulerLocation(String topologyName) {
   }
 
   /**
+   * Clean all states of a heron topology. This goes through each piece of state that needs
+   * to be cleaned up and will log out a warning if it could not be cleaned up properly.
+   * TMasterLocation, PackingPlan, PhysicalPlan, SchedulerLocation, ExecutionState, and Topology
+   * @param topologyName the name of the topology that we should clean the state for
+   */
+  public void cleanState(String topologyName) {
```

Comment:
We want to guard against new locations being added that are not included in this block. Now that we have the enum [FileSystemStateManager.StateLocation](https://github.com/twitter/heron/blob/f0472c3932f839657fd98e6d3f85c7e87f7b0982/heron/statemgrs/src/java/com/twitter/heron/statemgr/FileSystemStateManager.java#L42) we could iterate over all enums with a case statement for deleting each. If the case statement misses we could throw an exception, which would let us know that a new one was added but not included.

If we do that though, we might want to push this method into the `IStateManager` interface though. Thoughts?